content_handler/accessibility_bridge.cc

```diff
@@ -0,0 +1,79 @@
+// Copyright 2017 The Fuchsia Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/content_handler/accessibility_bridge.h"
+
+#include <unordered_set>
+
+#include "apps/maxwell/services/context/context_writer.fidl.h"
+#include "apps/modular/lib/rapidjson/rapidjson.h"
+#include "flutter/lib/ui/semantics/semantics_node.h"
+#include "lib/app/cpp/application_context.h"
+#include "lib/fxl/macros.h"
+
+namespace flutter_runner {
+
+AccessibilityBridge::AccessibilityBridge(app::ApplicationContext* context)
+    : writer_(context->ConnectToEnvironmentService<maxwell::ContextWriter>()) {
+  writer_.set_connection_error_handler(
+      [] { FXL_LOG(ERROR) << "Error connecting to ContextWriter"; });
+  FXL_VLOG(2) << "Constructing AccessibilityBridge.";
+}
+
+void AccessibilityBridge::UpdateSemantics(
+    const std::vector<blink::SemanticsNode>& update) {
+  for (const auto& node : update) {
+    semantics_nodes_[node.id] = node;
+  }
+  std::vector<int> visited_nodes;
+  UpdateVisitedForNodeAndChildren(0, &visited_nodes);
+  EraseUnvisitedNodes(visited_nodes);
+
+  // The data sent to the Context Service is a JSON formatted list of strings
```

Comment:
What is contained in `label` is not necessary shown on screen. For example, an `IconButton` to open a menu might have the label "Open menu", which is not necessarily visible on screen.

Reply:
I updated the comment to be more explicit about this. Is it fair to say that the collection of labels is a superset of the on screen text?

Reply:
Not necessarily. For example, there might be decorative text on the screen that isn't included in the accessibility data.